src/Orleans/Configuration/MessagingConfiguration.cs

```diff
@@ -106,6 +108,11 @@ public interface IMessagingConfiguration
         /// The maximum batch size for automatic batching of messages, when message batching is used.
         /// </summary>
         int MaxMessageBatchingSize { get; set; }
+
+        /// <summary>
+        /// The list of serialization providers
+        /// </summary>
+        HashSet<Type> SerializationProviders { get; }
```

Comment:
Use TypeInfo. This move to fully avoid Type in flavor of TypeInfo is what I will do later. For now I just kept Type and called Type.GetTypeInfo() whenever is necessary but in this case, ou can already have HashSet<TypeInfo> there but keep in mind that if serialization is equired, you should use HashSet<string> instead.